### DISCUSSION TITLE
Pass classes to box wrapper (like spacing is done)

### DISCUSSION THREAD

**Discussion by anastasiauz** [OFFLINE-GmbH/oc-boxes-support#112] [Help] [answered]
Hi! Absolutely love this plugin and now trying to create a very flexible builder using UI Kit css. The idea is to nest a row box -> column box -> title, content, other pices... but as every box needs to be wrapped by a div with service boxes classes (oc-*) it's not possible to have: grid (row) div > column div (direct ansector). For frontend BOXES_RENDER_SCAFFOLDING=false helps but then in backend preview it obviously stops working. Is there a way to pass these classes to views/_box template along with spacing classes as box object properties? I hardcoded that in the view and it completely solved my problem, so now the question is how to do that with events or some other way. Please suggest some solution.

**Comment by tobias-kuendig** [comment]
`display: contents` comes to mind, that you could add to the `.oc-box` class.  This way, the following would be possible:

```html
<style>
   .oc-box { display: contents; }
</style>

<div class="grid">
    <div class="oc-box">  <!-- This will be ignored in the grid -->
        <div class="some-grid-item">
```

As a more general solution, we could add an Event that you could use like so:

```php
Event::listen('offline.boxes.extendBoxClasses', function (Box $box) {
    if ($box->config->handle === 'some-grid-item') {
        return 'your-custom-class';
    }
})
```

Would that solution be suitable for your use-case?

**Reply by anastasiauz** [reply]
Tobias, thank you for the fast answer! Indeed, .oc-box { display: contents; } could be a good simple solution, but then .oc-boxes-wrapper in editor view starts to 'float' - it is in grid now too... I have columns 33% / 66% and they still don't look like columns.

Injecting custom classes to the wrapper would be a nice flexible solution, and likely useful not only for my case.

Thank you and have a great weekend!

**Comment by tobias-kuendig** [comment, marked as answer]
This feature is now available in v3.5.5, thank you for the input!

[Docs](https://docs.boxes.offline.ch/reference/events.html#offline-boxes-extendboxscaffoldingclasses)

**Comment by anastasiauz** [comment]
Tobias, thanks a lot!!!

**Comment by anastasiauz** [comment]
Have you any idea why classes are 'leaking' into child boxes? This is unwanted behaviour

**Reply by tobias-kuendig** [reply]
Hm, yes. They are added to the context and therefore also available for all children. Please update to v3.5.7, it should be fixed there.

**Reply by anastasiauz** [reply]
Thank you, now everything is perfect. And many-many thanks for speedy replies and fixes, that's amazing!
